include/exadg/incompressible_navier_stokes/spatial_discretization/calculators/wall_shear_stress_calculator.cpp

```diff
@@ -0,0 +1,232 @@
+          {
+            face_cell_index_matches[i] = j;
+            break;
+          }
+        }
+      }
+
+      face_to_cell_index[face] = face_cell_index_matches;
+    }
+
+    break;
+  }
+}
+
+template<int dim, typename Number>
+void
+WallShearStressCalculator<dim, Number>::compute_wall_shear_stress(
+  VectorType &                                  dst,
+  VectorType const &                            src,
+  std::shared_ptr<dealii::Mapping<dim> const>   mapping,
+  std::vector<dealii::types::boundary_id> const write_wall_shear_stress_on_IDs) const
+{
+  dst = 0;
+
+  // fill boundary ID to bool map OR write on all boundaries
+  AssertThrow(
+    write_wall_shear_stress_on_IDs.size() > 0,
+    dealii::ExcMessage(
+      "Provide at least write_wall_shear_stress_on_IDs = {dealii::numbers::invalid_boundary_id}.")) bool
+    write_on_all_boundary_IDs = false;
+  if(write_wall_shear_stress_on_IDs.size() == 1)
+  {
+    if(write_wall_shear_stress_on_IDs[0] == dealii::numbers::invalid_boundary_id)
+      write_on_all_boundary_IDs = true;
+  }
+  std::map<dealii::types::boundary_id, bool> write_on_boundary_IDs;
+  for(auto const id : write_wall_shear_stress_on_IDs)
+    write_on_boundary_IDs.insert({id, true});
+
+  unsigned int const dofs_per_component =
+    matrix_free->get_dof_handler(dof_index).get_fe(0).dofs_per_cell / dim;
+  unsigned int const    fe_degree = matrix_free->get_dof_handler(dof_index).get_fe(0).degree;
+  dealii::FESystem<dim> fe_system(dealii::FE_DGQ<dim>(fe_degree), dim);
+
+  dealii::QGaussLobatto<dim - 1> face_quadrature(fe_degree + 1);
+  dealii::QGaussLobatto<dim>     quadrature(fe_degree + 1);
+
+#ifdef DEBUG
+  dealii::FEFaceValues<dim> fe_face_values(*mapping,
+                                           fe_system,
+                                           face_quadrature,
+                                           dealii::update_quadrature_points |
+                                             dealii::update_gradients |
+                                             dealii::update_normal_vectors);
+  dealii::FEValues<dim>     fe_values(*mapping,
+                                  fe_system,
+                                  quadrature,
+                                  dealii::update_quadrature_points);
+#else
+  dealii::FEFaceValues<dim> fe_face_values(
+    *mapping, fe_system, face_quadrature, dealii::update_gradients | dealii::update_normal_vectors);
+#endif
+
+  std::vector<dealii::Tensor<2, dim>> velocity_gradients(fe_face_values.n_quadrature_points);
+  std::vector<dealii::types::global_dof_index> dof_indices(fe_system.dofs_per_cell);
+  const dealii::FEValuesExtractors::Vector     vector(0);
+
+  // fill vector with entries on boundary via cell-loop or prepared container.
+  for(auto const & cell : matrix_free->get_dof_handler(dof_index).active_cell_iterators())
+  {
+    if(cell->is_locally_owned() == false)
+      continue;
+
+    if(cell->at_boundary() == false)
+      continue;
```

Comment:
Personally, I do not like the `if(.... == false) continue;` style (negations are always harder to read for humans than positive statements). I think this "style" originates from indentation aspects with 80 characters per line in deal.II and the fact that deal.II indentation introduces two tabs whenever using brackets. Note that all this does not apply for ExaDG.

Reply:
Ok, I am fine with that style!

Reply:
Same style question: should I even write `if( ... == true)`?

Reply:
I would skip `==true`

Reply:
passt, then I will adopt that style from now on!